src/main/java/md2k/mCerebrum/cStress/features/ECGFeatures.java

```diff
@@ -582,13 +578,11 @@ public DataPoint rr_ave_update(List<DataPoint> rpeak_temp1, DataPointStream rr_a
                 for (int i = peak_interval.size() - AUTOSENSE.PEAK_INTERVAL_MINIMUM_SIZE; i < peak_interval.size(); i++) {
                     result.value += peak_interval.get(i);
                 }
-                result.value /= 8.0;
-                result.timestamp = rpeak_temp1.get(rpeak_temp1.size() - 1).timestamp;
-                rr_ave.add(result);
+                rr_ave.add(new DataPoint(rpeak_temp1.get(rpeak_temp1.size() - 1).timestamp, result.value /= 8.0));
```

Comment:
I'm not sure I follow.  The old code was doing that too.
But the old code had two bugs:
a).  It was return the datapoint (0,0.0) when peak_interval.size() < AUTOSENSE.PEAK_INTERVAL_MINIMUM_SIZE, when it should be returning the old rr_ave
and b) it was skipping the first difference.